src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilderData.cs

```diff
@@ -2,193 +2,51 @@
-            {
-                // enlarge the arrays
-                byte[][] temp = new byte[m_iCAs * 2][];
-                ConstructorInfo[] tempCon = new ConstructorInfo[m_iCAs * 2];
-                for (int i = 0; i < m_iCAs; i++)
-                {
-                    temp[i] = m_CABytes[i];
-                    tempCon[i] = m_CACons[i];
-                }
-                m_CABytes = temp;
-                m_CACons = tempCon;
-            }
-
-            byte[] attrs = new byte[binaryAttribute.Length];
-            Buffer.BlockCopy(binaryAttribute, 0, attrs, 0, binaryAttribute.Length);
-            m_CABytes[m_iCAs] = attrs;
-            m_CACons[m_iCAs] = con;
-            m_iCAs++;
-        }
+        internal void AddModule(ModuleBuilder dynModule) => _moduleBuilderList.Add(dynModule);
         
         /// <summary>
         /// Helper to ensure the type name is unique underneath assemblyBuilder.
         /// </summary>
         internal void CheckTypeNameConflict(string strTypeName, TypeBuilder enclosingType)
         {
-            for (int i = 0; i < m_moduleBuilderList.Count; i++)
+            for (int i = 0; i < _moduleBuilderList.Count; i++)
             {
-                ModuleBuilder curModule = m_moduleBuilderList[i];
+                ModuleBuilder curModule = _moduleBuilderList[i];
                 curModule.CheckTypeNameConflict(strTypeName, enclosingType);
             }
-
-            // Right now dynamic modules can only be added to dynamic assemblies in which
-            // all modules are dynamic. Otherwise we would also need to check loaded types.
-            // We only need to make this test for non-nested types since any
-            // duplicates in nested types will be caught at the top level.
-            //      if (enclosingType == null && m_assembly.GetType(strTypeName, false, false) != null)
-            //      {
-            //          // Cannot have two types with the same name
-            //          throw new ArgumentException(SR.Argument_DuplicateTypeName);
-            //      }
         }
 
-        internal List<ModuleBuilder> m_moduleBuilderList;
-        internal List<ResWriterData> m_resWriterList;
-        internal string m_strAssemblyName;
-        internal AssemblyBuilderAccess m_access;
-        private InternalAssemblyBuilder m_assembly;
-
-        internal Type[] m_publicComTypeList;
-        internal int m_iPublicComTypeCount;
+        internal List<ModuleBuilder> _moduleBuilderList;
+        internal AssemblyBuilderAccess _access;
+        private readonly InternalAssemblyBuilder _assembly;
 
-        internal bool m_isSaved;
-        internal const int m_iInitialSize = 16;
+        internal const int InitialSize = 16;
 
         // hard coding the assembly def token
-        internal const int m_tkAssembly = 0x20000001;
+        internal const int AssemblyDefToken = 0x20000001;
 
         // tracking AssemblyDef's CAs for persistence to disk
```

Comment:
This comment is not relevant for `_entryPointMethod`

Reply:
fixed